playbooks/collect_activity/stats.py

```diff
@@ -26,54 +48,50 @@
 
-    # Produce the paths to all of the "access.log*" files in the Nginx log directory to be parsed.
-    log_files = [
-        f for f in os.listdir(LOG_PATH)
-        if os.path.isfile(os.path.join(LOG_PATH, f)) and f.startswith('access.log')
-    ]
-
-    # Walk through all of the Nginx logs, storing the found remote host IP addresses in a set to enforce uniqueness.
-    unique_hits = set()
-    for file in log_files:
-        print >> sys.stderr, 'Parsing log file: {file}'.format(file=file)
-
-        file_path = os.path.join(LOG_PATH, file)
-        # Make sure we use gzip to decompress any compressed log files.
-        if file.endswith('.gz'):
-            handle = gzip.open(file_path, 'rb')
-        else:
-            handle = open(file_path, 'r')
-
-        # Run each access log line through a regex to extract the IPv4 addresses of remote hosts.
-        for line in handle:
-            match = re.match(NGINX_ACCESS_PATTERN, line)
-            if match:
-                unique_hits.add(match.group('ipaddress'))
-        handle.close()
-
     ms = modulestore()
 
     stats = {
         'users': User.objects.count(),
         'courses': len(ms.get_courses()),
-        'hits': len(unique_hits),
     }
 
-    # Grab the hostname address address passed to this script by the ansible playbook. This will be used as the
-    # ConfigParser "section" name for the output data. This is so the data can easily be associated with this app
-    # server by the `activity_csv` management command.
-    public_ip = sys.argv[1]
+    if not args.skip_hit_statistics:
+        # Produce the paths to all of the "access.log*" files in the Nginx log directory to be parsed.
+        log_files = [
+            f for f in os.listdir(LOG_PATH)
+            if os.path.isfile(os.path.join(LOG_PATH, f)) and f.startswith('access.log')
+        ]
+
+        # Walk through all of the Nginx logs, storing the found remote host IP addresses in a set to enforce uniqueness.
+        unique_hits = set()
+        for file in log_files:
+            print >> sys.stderr, 'Parsing log file: {file}'.format(file=file)
+
+            file_path = os.path.join(LOG_PATH, file)
+            # Make sure we use gzip to decompress any compressed log files.
+            if file.endswith('.gz'):
+                handle = gzip.open(file_path, 'rb')
+            else:
+                handle = open(file_path, 'r')
+
+            # Run each access log line through a regex to extract the IPv4 addresses of remote hosts.
+            for line in handle:
+                match = re.match(NGINX_ACCESS_PATTERN, line)
+                if match:
+                    unique_hits.add(match.group('ipaddress'))
+            handle.close()
+
+        stats['hits'] = len(unique_hits)
 
     # Build the ConfigParser data.
     config = ConfigParser()
-    config.add_section(public_ip)
+    config.add_section(args.config_section)
     for key, value in stats.items():
         config.set(public_ip, key, value)
```

Comment:
@pcockwell, public_ip is not defined before as you have removed it.

Reply:
Done in next commit